libsplinter/src/auth/oauth/builder/openid.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use reqwest::blocking::Client;
+
+use crate::auth::oauth::{builder::OAuthClientBuilder, error::OAuthClientBuildError, OAuthClient};
+use crate::auth::rest_api::identity::{openid::OpenIdUserIdentityProvider, IdentityProvider};
+use crate::error::{InternalError, InvalidStateError};
+
+/// Builds a new `OAuthClient` using an OpenID discovery document.
+///
+/// # Arguments
+///
+/// * `client_id` - The OpenID provider OAuth client ID
+/// * `client_secret` - The OpenID provider OAuth client secret
+/// * `redirect_url` - The endpoint that the OpenID provider will redirect to after it has completed
+///   authorization
+/// * `discovery_url` - The OpenID provider's discovery document
+pub struct OpenIdOAuthClientBuilder {
+    openid_discovery_url: Option<String>,
+    inner: OAuthClientBuilder,
+}
+
+impl OpenIdOAuthClientBuilder {
+    pub fn new() -> Self {
```

Comment:
Missing a doc comment. Also, maybe this should be private if it's only constructed using the `OAuthClientBuilder::new_openid` method? If you make this private, do the same for the GitHub builder.

Reply:
It can't be private, as the parent module needs to access it.  `pub (super)`, maybe?